### PR TITLE
Remove Banner, CollapsiblePanel, PromoBanner and TextArea React components

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Chromatic

https://1016-delete-react--60f9b557105290003b387cd5.chromatic.com

## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1016
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/34437

## Acceptance criteria
- [x] Banner, CollapsiblePanel, PromoBanner and TextArea are removed from component-library

## Definition of done
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by andresriveratoro** [comment]
Do we want to make this a breaking change?

File: packages/react-components/src/components/CollapsiblePanel/CollapsiblePanel.jsx [hunk 1]
```diff
@@ -1,102 +0,0 @@
-import PropTypes from 'prop-types';
-import React from 'react';
-import Scroll from 'react-scroll';
```
Comment (bkjohnson) [inline packages/react-components/src/components/CollapsiblePanel/CollapsiblePanel.jsx:3]: With this happening I think we can remove `react-scroll` and close https://github.com/department-of-veterans-affairs/va.gov-team/issues/34437.

File: packages/react-components/src/components/CollapsiblePanel/static-markup.stories.js [hunk 2]
```diff
@@ -1,60 +0,0 @@
-import ReactHtmlParser from 'react-html-parser';
```
Comment (bkjohnson) [inline packages/react-components/src/components/CollapsiblePanel/static-markup.stories.js:1]: We can probably remove this dependency as well.

**Review by bkjohnson — COMMENTED** [review]
Left a few comments on dependency cleanup.

And yes, I would say this is a breaking change.

**Review by bkjohnson — APPROVED** [review]
This looks good :rocket:

**Comment by bkjohnson** [comment]
We should probably remove [this piece of config in `vets-website`](https://github.com/department-of-veterans-affairs/vets-website/blob/b791ebe1d3e5a7bdd66aeb4c8a44dc9c38d6361a/.babelrc#L66) for the sake of tidying up.

**Pull request merged**